src/Symfony/Component/HttpKernel/Tests/EventListener/CacheAttributeListenerTest.php

```diff
@@ -91,6 +91,61 @@ public function testResponseIsPrivateIfConfigurationIsPublicFalse()
         $this->assertTrue($this->response->headers->hasCacheControlDirective('private'));
     }
 
+    public function testResponseIsPrivateIfSharedMaxAgeSetAndNoStoreIsTrue()
+    {
+        $request = $this->createRequest(new Cache(smaxage: 1, noStore: true));
+
+        $this->listener->onKernelResponse($this->createEventMock($request, $this->response));
+
+        $this->assertFalse($this->response->headers->hasCacheControlDirective('public'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('private'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('no-store'));
+    }
+
+    public function testResponseIsPublicIfConfigurationIsPublicTrueNoStoreFalse()
+    {
+        $request = $this->createRequest(new Cache(public: true, noStore: false));
+
+        $this->listener->onKernelResponse($this->createEventMock($request, $this->response));
+
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('public'));
+        $this->assertFalse($this->response->headers->hasCacheControlDirective('private'));
+        $this->assertFalse($this->response->headers->hasCacheControlDirective('no-store'));
+    }
+
+    public function testResponseIsPrivateIfConfigurationIsPublicTrueNoStoreTrue()
+    {
+        $request = $this->createRequest(new Cache(public: true, noStore: true));
+
+        $this->listener->onKernelResponse($this->createEventMock($request, $this->response));
+
+        $this->assertFalse($this->response->headers->hasCacheControlDirective('public'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('private'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('no-store'));
+    }
+
+    public function testResponseIsPrivateNoStoreIfConfigurationIsNoStoreTrue()
+    {
+        $request = $this->createRequest(new Cache(noStore: true));
+
+        $this->listener->onKernelResponse($this->createEventMock($request, $this->response));
+
+        $this->assertFalse($this->response->headers->hasCacheControlDirective('public'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('private'));
+        $this->assertTrue($this->response->headers->hasCacheControlDirective('no-store'));
+    }
+
+    public function testResponseIsPrivateNoStoreIfConfigurationSmaxAgeIsSetAndNoStoreIsTrue()
+    {
+        $request = $this->createRequest(new Cache(smaxage: 0, noStore: true));
```

Comment:
This test seems to be a duplicate of `testResponseIsPrivateIfSharedMaxAgeSetAndNoStoreIsTrue`. Only the value of `smaxage` is different but there is no assertion on it.

Reply:
Yeah i mixed things with these method names sorry